File: backend/internal/domain/master_exercise_example.go
package domain

import "time"

// MasterExerciseExample は MasterExercise に紐付く入力例 / 期待出力例の 1 ペア。
// 1 問に複数ケースを持ち、表示 / 採点順序は OrderIndex で安定ソートする。
type MasterExerciseExample struct {
	ID uint64 `json:"id"`
	// (exercise_id, order_index) の UNIQUE で同一問題内の OrderIndex 衝突を DB レベルで弾く。
	ExerciseID uint64 `json:"exerciseId"`
	// DEFAULT を持たせない（default:0 だと未指定 INSERT が 0 で衝突するため）。
	OrderIndex     int16     `json:"orderIndex"`
	InputText      string    `json:"inputText"`
	ExpectedOutput string    `json:"expectedOutput"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

File: backend-java/src/main/java/com/normanblog/frestyle/entity/MasterExerciseExample.java
package com.normanblog.frestyle.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** MasterExercise に紐付く入力例 / 期待出力例の 1 ペア。表示 / 採点順序は orderIndex で安定ソート。 */
@Entity
@Table(name = "master_exercise_examples")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MasterExerciseExample {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "exercise_id", nullable = false)
  private Long exerciseId;

  @Column(name = "order_index", nullable = false)
  private short orderIndex;

  @Column(name = "input_text", columnDefinition = "text", nullable = false)
  private String inputText;

  @Column(name = "expected_output", columnDefinition = "text", nullable = false)
  private String expectedOutput;

  @Column(name = "created_at")
  private Instant createdAt;

  @Column(name = "updated_at")
  private Instant updatedAt;
}
